code/modules/mob/living/simple_animal/bot/medbot.dm

```diff
@@ -516,61 +557,70 @@ GLOBAL_VAR(medibot_unique_id_gen)
 			soft_reset()
 			break
 
-		var/treatment_method = null
-
-		if(C.getBruteLoss() >= heal_threshold)
-			treatment_method = BRUTE
 
-		else if(C.getFireLoss() >= heal_threshold)
-			treatment_method = BURN
+		var/treat_behaviour = "internal_beaker"
+		if(!treat_behaviour && reagent_glass && reagent_glass.reagents.total_volume)
+			for(var/datum/reagent/R in reagent_glass.reagents.reagent_list)
+				if(C.reagents.has_reagent(R.type))
+					treat_behaviour = null
 
-		else if(C.getOxyLoss() >= (5 + heal_threshold))
-			treatment_method = OXY
+		if(emagged) //Emagged! Time to drain everybody.
+			treat_behaviour = "suck"
 
-		else if(C.getToxLoss() >= heal_threshold)
-			treatment_method = TOX
-
-		if(!treatment_method && emagged != 2) //If they don't need any of that they're probably cured!
+		if(!treat_behaviour) //If they don't need any of that they're probably cured!
 			if(C.maxHealth - C.health < heal_threshold)
-				to_chat(src, "<span class='notice'>[C] is healthy! Your programming prevents you from injecting anyone without at least [heal_threshold] damage of any one type ([heal_threshold + 5] for oxygen damage.)</span>")
+				to_chat(src, "<span class='notice'>[C] is healthy! Your programming prevents you from injecting anyone without at least [heal_threshold] damage of any one type ([heal_threshold + 15] for oxygen damage.)</span>")
 			var/list/messagevoice = list("All patched up!" = 'sound/voice/medbot/patchedup.ogg',"An apple a day keeps me away." = 'sound/voice/medbot/apple.ogg',"Feel better soon!" = 'sound/voice/medbot/feelbetter.ogg')
 			var/message = pick(messagevoice)
-			speak(message)
+			speak(message, radio_channel)
 			playsound(src, messagevoice[message], 50)
 			bot_reset()
-			tending = FALSE
-		else if(patient)
-			C.visible_message("<span class='danger'>[src] is trying to tend the wounds of [patient]!</span>", \
-				"<span class='userdanger'>[src] is trying to tend your wounds!</span>")
-
-			if(do_after(src, 2 SECONDS, patient)) //Slightly faster than default tend wounds, but does less HPS
-				if((get_dist(src, patient) <= 1) && (on) && assess_patient(patient))
-					var/healies = heal_amount
-					var/obj/item/storage/firstaid/FA = firstaid
-					if(treatment_method == initial(FA.damagetype_healed)) //using the damage specific medkits give bonuses when healing this type of damage.
-						healies *= 1.5
-					if(treatment_method == TOX && HAS_TRAIT(patient, TRAIT_TOXINLOVER))
-						healies *= -1.5
-					if(emagged == 2)
-						patient.reagents.add_reagent(/datum/reagent/toxin/chloralhydrate, 5)
-						patient.apply_damage_type((healies*1),treatment_method)
-						log_combat(src, patient, "pretended to tend wounds on", "internal tools", "([uppertext(treatment_method)]) (EMAGGED)")
+			return
+
+		C.visible_message("<span class='danger'>[src] is trying to inject [patient]!</span>", "<span class='userdanger'>[src] is trying to inject you!</span>")
```

Comment:
after this line to the end of the code, you need to rewrite codes.
* Try not to use multiple `if` wraps. Multiple `if` wrapping makes the code unreadable.
* Try to **_Early Return_** technic. This reduces `if` usages.
  * The line 492 is the example of **_Early Return_** (That I suggested)